### DISCUSSION TITLE
🍺 Old homebrew discussion

### DISCUSSION THREAD

**Discussion by pakerwreah** [pakerwreah/Calendr#185] [General]
[![release](https://img.shields.io/github/v/release/pakerwreah/Calendr)](https://github.com/pakerwreah/Calendr/releases/latest)
1. Go to the releases page and download the most recent `Calendr.zip`.
2. Unzip the archive and drag `Calendr.app` to the `Applications` folder.
3. ~~Right click it and select `Open`.~~

The last step is not needed anymore. The app is now signed. 🔏

**Comment by mracos** [comment]
For reference in case someone wants to install with homebrew [here is how I did it](https://github.com/mracos/homebrew-tap/blob/a5c3aca2453329c5e937d603fd0f6d73c81dacf4/Casks/calendr.rb) 😄

**Comment by pakerwreah** [comment]
@mracos 👏🏻 nice!
Is this tied to a specific version? 🤔

**Comment by mracos** [comment]
Initially it is, but it _should_ fetch latest releases [because of the `appcast` stanza on it](https://github.com/orgs/Homebrew/discussions/173) on new installs

I might try to submit a PR to [homebrew-cask](https://github.com/Homebrew/homebrew-cask) later

**Comment by mracos** [comment]
Just a FYI, [got rejected](https://github.com/Homebrew/homebrew-cask/pull/152677#issuecomment-1672506377) because the app needs to be signed which I have no idea if feasible or possible. Will keep using the one in my own tap for now 😄

Thank you so much for this little app 💙

**Comment by pakerwreah** [comment]
It surely can be signed, I just don't want to pay Apple $99 per year 😆

**Comment by mracos** [comment]
That's... understandable 🤣

What a bummer that you need to pay a subscription to sign stuff. Well, thanks for the explanation and the TIL!

**Comment by thedavidweng** [comment]
> Just a FYI, [got rejected](https://github.com/Homebrew/homebrew-cask/pull/152677#issuecomment-1672506377) because the app needs to be signed which I have no idea if feasible or possible. Will keep using the one in my own tap for now 😄仅供参考，被拒绝是因为该应用程序需要签名，我不知道这是否可行或可能。现在😄将继续使用我自己的水龙头中的那个
> 
> Thank you so much for this little app 💙非常感谢你这个小应用程序 💙

Is it possible to have a github action that automatically push the latest version to the tap?

**Comment by pakerwreah** [comment]
@mracos I think you said your tap was not tied to a specific version, right? 🤔

btw, I was searching for information on this and I'm seeing this warning in a lot of places:
`Calling the appcast stanza is deprecated! Use the livecheck stanza instead`

ps.: I think macOS "compress" task stopped adding the `.app` to the zip file name 👀

**Comment by mracos** [comment]
> Is it possible to have a github action that automatically push the latest version to the tap?

@thedavidweng It could checks once a day and then submit a PR? Maybe something already exists on the marketplace. Feel free o contribute if you want!

Also, @pakerwreah, do you want to move this formula to a repo under your wing?

> @mracos I think you said your tap was not tied to a specific version, right? 🤔

yeah @pakerwreah, I shared a bit of fake news, my bad on that 😅
I assumed that it was not bc of the appcast stanza but it seems (at [least with livecheck](https://docs.brew.sh/Brew-Livecheck)) that we have to manually run `brew livecheck`

<img width="562" alt="Screenshot 2024-03-29 at 16 24 36" src="https://github.com/pakerwreah/Calendr/assets/14015177/b5a5916a-2cfd-4f10-9140-7cdbd2c92730">

And tbh it was not working. Fixed it, also fixed the `appcast` warning (although I thought I [fixed it a while ago](https://github.com/mracos/homebrew-tap/commit/4310ef4ddb991de223ea1a97433bec119e7d864f) but missed a dangling call), and [already updated](https://github.com/mracos/homebrew-tap/blob/f9a7fbf1c4ffdb43d24bdac7caec0d49cd81b042/Casks/calendr.rb) to the latest version!

Thanks for the heads up 🙇

**Comment by thedavidweng** [comment]
Thank you! Yes I do believe this can be done with a github action, @pakerwreah do you want to maintain your own tap?

**Comment by pakerwreah** [comment]
I don't want another thing to maintain 😅
I can see myself, in the future, losing hours trying to figure out some random homebrew breaking change.

**Comment by pakerwreah** [comment]
@mracos The app is now signed, notarized and available via my own cask 🎉 
Thanks for your help. 🙇🏻‍♂️ 

New installation instructions: https://github.com/pakerwreah/Calendr/issues/217

Oh, and the updates are automated via Github Actions ⚙️

ps.: I'm not sure about deploying it to the official repo, because then I have to deal with PRs for every update.
I recently had to rollback a version because it was crashing. I can imagine the anxiety attack I would have with waiting for someone to merge my PR. 🙈

But feel free to change my mind. 😅

**Comment by mracos** [comment]
So cool @pakerwreah!!! Amazing to hear 💙 

> ps.: I'm not sure about deploying it to the official repo, because then I have to deal with PRs for every update.

You could also automate creating the PR on the official repo
I think one can do it by running `brew bump` on a github action cronjob, kinda like I did it for my [personal cask](https://github.com/mracos/homebrew-tap/blob/master/.github/workflows/bump.yml) (without the `--tap` maybe?)

> I recently had to rollback a version because it was crashing. I can imagine the anxiety attack I would have with waiting for someone to merge my PR. 🙈

From what I've seen, people that maintain homebrew are quite responsive.
That with a minor patch bump reverting plus automations that create the PR automatically could mitigate this feeling?

Although, _for me_, the `tap` existing on the same "organization" as the software already makes it official enough so no strong opinions on this haha

**Reply by pakerwreah** [reply]
yeah, there's a `brew bump-cask-pr` to do everything (i.e. forking, committing, pushing)
but I also agree with you that a tap owned by me is official enough 😄
